bubblewrap.c

```diff
@@ -1491,6 +1510,14 @@ main (int    argc,
         }
 
       /* Initial launched process, wait for exec:ed command to exit */
+      if (opt_info_fd != -1)
+        {
+          char output[64];
+          int len = sprintf (output, "{\n    \"child-pid\": %i\n}\n", pid);
+          if (write (opt_info_fd, output, len) != len)
+            die_with_error ("Write to info_fd");
+          close (opt_info_fd);
+        }
```

Comment:
I was hoping we could use something like SO_PEERCRED on e.g. the block_fd so that we can rely on the kernel to pass the info, rather than having to serialize it like this. But I don't think that will work, as the creds on the fd you passed to the child will probably be your own (the socket creds are stashed pretty early during socket creation).

Any way, this is pretty tightly coupled with the block fd, and could be done using only one fd. Maybe we can combine them? I.e. just always write the info to the fd? As long as that doesn't block (for instance, if the data is smaller than the normal pipe buffer) it should be safe to ignore the info and just wait for the fd to close, no?